src/SonarScanner.MSBuild.Common/ConsoleLogger.cs

```diff
@@ -137,6 +138,31 @@ public void WriteUIWarnings(string outputFolder)
         }
     }
 
+    public void AddTelemetryMessage(string key, string value) =>
+        telemetryMessages.Add(new(key, value));
+
+    public void WriteTelemetry(string outputFolder)
+    {
+        var telemetryMessagesJson = new StringBuilder();
+        foreach (var message in telemetryMessages)
+        {
+            var entry = new JObject();
+            entry[message.Key] = message.Value;
```

Comment:
We could do `AddTelemetryMessage(string key, object value)`. The you can do
```suggestion
            entry[message.Key] = JToken.FromObject(message.Value);
```

That way we can create 
```js
{ key: true }
{ key: 42 }
{ key: 3.14 }
{ key: "Hello" }
```

That may help later with the aggregation in the plugin because we do not have to go string -> number in the plugin. It looks easy enough to do here and allows for a bit more flexibility in the plugin.

Note: I convert everything to string in the plugin at the moment, but I do support all json types (string, numbers, and boolean)

Note2: We need to raise an exception, if `JToken.FromObject` does not return a `JValue`as we do not supported nested jsons.